python/pyarrow/tests/test_table.py

```diff
@@ -1286,6 +1290,30 @@ def test_table_add_column():
     assert t4.equals(expected)
 
 
+def test_record_batch_add_column():
```

Comment:
We could maybe parametrize the existing test for Table instead. 

If you search for the `@pytest.mark.parametrize` cases already in this file, you will see some other examples

Reply:
https://github.com/apache/arrow/pull/39506/commits/559f86a3264b4427132eafef4c2651eee55517a7